test/fast/Distributions.test.ts

```diff
@@ -1,22 +1,176 @@
+import { Issuer, Marketplace, Distribution, Distribution__factory, IERC20, Fast, FastTokenFacet } from "../../typechain";
+import { abiStructToObj, DistributionPhase } from "../utils";
+import { fastFixtureFunc } from "../fixtures/fast";
 
 chai.use(solidity);
 chai.use(smock.matchers);
 
-describe("Distributions", () => {
-  before(async () => { });
-  beforeEach(async () => { });
+describe.only("Distributions", () => {
+  let deployer: SignerWithAddress,
+    issuerMember: SignerWithAddress,
+    governor: SignerWithAddress,
+    alice: SignerWithAddress;
+
+  let issuer: FakeContract<Issuer>,
+    marketplace: FakeContract<Marketplace>,
+    fast: Fast,
+    token: FastTokenFacet,
+    erc20: FakeContract<IERC20>;
+
+  const fastDeployFixture = deployments.createFixture(fastFixtureFunc);
+
+  before(async () => {
+    // Keep track of a few signers.
+    [deployer, issuerMember, governor, alice] = await ethers.getSigners();
+
+    // Mock an Issuer and an Marketplace contract.
+    issuer = await smock.fake("Issuer");
+    marketplace = await smock.fake("Marketplace");
+    marketplace.issuerAddress.returns(issuer.address);
+
+    marketplace.isMember.whenCalledWith(governor.address).returns(true);
+    marketplace.isMember.returns(false);
+    marketplace.isActiveMember.whenCalledWith(governor.address).returns(true);
+    marketplace.isActiveMember.returns(false);
+
+    // TODO: Totally forgotten about what I'm doing... I just want a generic ERC20.
+    erc20 = await smock.fake("IERC20");
+  });
+
+  beforeEach(async () => {
+    await fastDeployFixture({
+      opts: {
+        name: "FastTokenFixture",
+        deployer: deployer.address,
+        afterDeploy: async (args) => {
+          ({ fast } = args);
+          token = await ethers.getContractAt<FastTokenFacet>("FastTokenFacet", fast.address);
+        },
+      },
+      initWith: {
+        issuer: issuer.address,
+        marketplace: marketplace.address,
+        governor: governor.address,
+      },
+    });
+
+  });
 
   describe("constructor", async () => {
-    it("ensures that the latched block is not in the future");
-    it("stores its initial parameters");
-    it("initializes the `available` funds using the initial `total`");
-    it("also stores the creation block");
-    it("delegates a call to balanceOf to the ERC20 token with the correct parameters");
-    it("makes sure that the available balance is exactly equal to the total");
-    it("advances to the FeeSetup phase");
+    describe("with the correct params passed", async () => {
+      let distribution: MockContract<Distribution>;
+
+      beforeEach(async () => {
+        // Stub out the ERC20 contract.
+        erc20.balanceOf.reset();
+        erc20.balanceOf.returns(10);
+
+        // Deploy the Distribution contract.
+        const factory = await smock.mock<Distribution__factory>("Distribution", deployer);
+
+        distribution = await factory.deploy({
```

Comment:
Given that this is a bit expensive, you could have a fixture (you know, the whole snapshot thingy that "memoizes" the state of the chain between tests.
But at the same time, it should't be too bad since this is not a diamond... So let it be for now :)